mlflow/store/artifact/mlflow_artifacts_repo.py

```diff
@@ -0,0 +1,79 @@
+from urllib.parse import urlparse
+import posixpath
+from collections import namedtuple
+
+from mlflow.store.artifact.http_artifact_repo import HttpArtifactRepository
+from mlflow.tracking._tracking_service.utils import get_tracking_uri
+from mlflow.exceptions import MlflowException
+
+
+def _parse_artifact_uri(artifact_uri):
+    ParsedURI = namedtuple("ParsedURI", "scheme host port path")
+    parsed_uri = urlparse(artifact_uri)
+    return ParsedURI(parsed_uri.scheme, parsed_uri.hostname, parsed_uri.port, parsed_uri.path)
+
+
+def _check_if_host_is_numeric(hostname):
+    if hostname:
+        try:
+            float(hostname)
+            return True
+        except ValueError:
+            return False
+    else:
+        return False
+
+
+def _validate_port_mapped_to_hostname(uri_parse):
+    # This check is to catch an mlflow-artifacts uri that has a port designated but no
+    # hostname specified. `urllib.parse.urlparse` will treat such a uri as a filesystem
+    # definition, mapping the provided port as a hostname value if this condition is not
+    # validated.
+    if uri_parse.host and _check_if_host_is_numeric(uri_parse.host) and not uri_parse.port:
+        raise MlflowException(
+            "The mlflow-artifacts uri was supplied with a port number: "
+            f"{uri_parse.host}, but no host was defined."
+        )
+
+
+class MlflowArtifactsRepository(HttpArtifactRepository):
+    """Scheme wrapper around HttpArtifactRepository for mlflow-artifacts server functionality"""
+
+    def __init__(self, artifact_uri):
+
+        super().__init__(self.resolve_uri(artifact_uri))
+
+    @classmethod
+    def resolve_uri(cls, artifact_uri):
+        tracking_uri = get_tracking_uri()
```

Comment:
Can we raise an exception if the scheme of `tracking_uri` is not `http` or `https` (e.g. `sqlite:///foo/bar`)?

Reply:
great idea :) added! I also simplified the parsing logic and temporarily set the tracking uri in the test function to validate against a valid server address (then reverted the global variable to not affect other tests with an inadvertent side-effect)

Reply:
Thanks for the update! Tracking URI looks like `http://localhost:5000` and doesn't contain `/api/2.0/mlflow-artifacts/artifacts`.

Reply:
updated the tests and cleaned all of that up

Reply:
Sorry I wasn't clear. The resolved artifact URI needs to have `/api/2.0/mlflow-artifacts/artifacts` (e.g. `http://localhost:5000/api/2.0/mlflow-artifacts/artifacts/path/to/dir`).